### PR TITLE
bd/board-factory

### PR DESCRIPTION
**Pull request by karolkolodziej**
- [x] Rename `Dashboards.Dashboard` into `Dashboard.Board`
- [x] Implement `Dashboard.board` factory
- [x] Fix demos
- [x] Fix documentation
- [x] change `xxx.dashboard` to `xxx.board`

### CODE REVIEW

**Review by sebastianbochan — APPROVED** [review]

**Review by goransle — APPROVED** [review]
LGTM

File: ts/masters/dashboards.src.ts [hunk 1]
```diff
@@ -53,9 +53,9 @@ declare global {
 const D = {
     ...Globals,
     ...Utilities,
-    _modules: (typeof _modules === 'undefined' ? {} : _modules),
+    Board: Dashboard,
+    board: Dashboard.board,
```
Comment (bre1470) [inline ts/masters/dashboards.src.ts:57]: The import should be renamed as well, so you can skip the colon thing.

File: samples/dashboards/demos/climate/demo.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ async function setupDashboard() {
     ).then(response => response.json());
     const mapPoints = await buildCitiesMap();
 
-    return new Promise(resolve => new Dashboards.Dashboard('container', {
+    return new Promise(resolve => new Dashboards.Board('container', {
```
Comment (bre1470) [inline samples/dashboards/demos/climate/demo.js:45]: As a reference demo this should have the `Dashboards.board` call.

File: docs/dashboards/your-first-dashboard.md [hunk 3]
```diff
@@ -29,9 +29,15 @@ Thus we have to import the main dashboard package, the Highcharts library and a
     * the div’s id where the dashboard should be placed (see point 1).
     * the object with dashboard options.
 
+    To do so, you can use factory function `Dashboards.board`:
 
     ```js
-        const dashboard = new Dashboards.Dashboard('container', {...})
+        Dashboards.board('container', {...})
+    ```
+    or the constructor to create a new instance of the dashboard:
```
Comment (bre1470) [inline docs/dashboards/your-first-dashboard.md:37]: or the class `Dashboards.Board`…

**Review by bre1470 — CHANGES_REQUESTED** [review]

**Review by pawellysy — APPROVED** [review]
LGTM! :D

**Review by bre1470 — CHANGES_REQUESTED** [review]
I do not see the point done:
- Rename Dashboards.Dashboard into Dashboards.Board

**Comment by karolkolodziej** [comment]
@bre1470 [isn't it that?](https://github.com/highcharts/highcharts/pull/18481/files#diff-af80584fc55104f51ff478328047f79b9630f972378d89ca846d93ecba4f1d8aR56)

**Comment by bre1470** [comment]
Okay, I understand now. I am expecting the rename of the file as well. So, from `ts/Dashboards/Dashboard.ts` to `ts/Dashboards/Board.ts` plus the adjustments of imports.

**Comment by bre1470** [comment]
@pawelfus
Should also `X.dashboard` properties (reference of `Board` class) be renamed?

**Comment by karolkolodziej** [comment]
IMO it will be cleaner and shorter if we use `xxx.board` instead of `xxx.dashboard` when referring to the `Board` instance.

**Review by bre1470 — CHANGES_REQUESTED** [review]
Cypress imports need adjustment.

**Comment by pawelfus** [comment]
> @pawelfus Should also `X.dashboard` properties (reference of `Board` instance) be renamed?

Yes, please do 👍 Note: I wouldn't agree to change like `series.chart` => `series.c`, because that would break existing implementations, but in this case - just checked with Karol docs and there's no place we recommend this, so it's safe 👍

**Review by bre1470 — APPROVED** [review]
Thank you! Looks good. 👍

**Pull request merged**